src/main/java/org/kizombadev/markdownparser/SyntaxAnalyzer.java

```diff
@@ -46,103 +73,117 @@ public Syntax parse(ImmutableList<Token> tokens) {
         return root;
     }
 
-    private void handleQuotation(RootSyntax currentRoot) {
-        Syntax quotation = new Quotation();
+    private void handleUnorderedList(MutableSyntax currentRoot) {
+        MutableSyntax unorderedList = new UnorderedListSyntax();
+
+        while (currentToken() == Token.Star && nextToken() == Token.Blank) {
+            stepTokenForward();
+            stepTokenForward();
+
+            handleUnorderedListItem(unorderedList);
+
+            if (currentToken() == Token.NewLine) {
+                stepTokenForward();
+            }
+        }
+
+        currentRoot.addChild(unorderedList);
+    }
+
+    private void handleUnorderedListItem(MutableSyntax unorderedList) {
+        MutableSyntax unorderedListItem = new UnorderedListItemSyntax();
+        handleLineContainer(unorderedListItem);
+        unorderedList.addChild(unorderedListItem);
+    }
+
+    private void handleQuotation(MutableSyntax currentRoot) {
+        MutableSyntax quotation = new QuotationSyntax();
         stepTokenForward();
 
         handleLineContainer(quotation);
         currentRoot.addChild(quotation);
     }
 
-    private void handleBigHeadlineLine(Syntax currentRoot) {
-        Syntax bigHeadline = new BigHeadline();
+    private void handleBigHeadlineLine(MutableSyntax currentRoot) {
+        MutableSyntax bigHeadline = new BigHeadline();
         stepTokenForward();
 
         handleLineContainer(bigHeadline);
         currentRoot.addChild(bigHeadline);
     }
 
-    private void handleSmallHeadlineLine(Syntax currentRoot) {
-        Syntax smallHeadline = new SmallHeadline();
+    private void handleSmallHeadlineLine(MutableSyntax currentRoot) {
+        MutableSyntax smallHeadline = new SmallHeadline();
         stepTokenForward();
 
 
         handleLineContainer(smallHeadline);
         currentRoot.addChild(smallHeadline);
     }
 
-    private void handleLineContainer(Syntax currentRoot) {
+    private void handleLineContainer(MutableSyntax currentRoot) {
 
         Token currentToken = currentToken();
 
         while (currentToken != null && !currentToken.equals(Token.NewLine)) {
 
-            if (currentToken.equals(Token.DoubleStar)) {
-                if (!stack.isEmpty() && stack.peek().equals(State.BOLD_BEGIN)) {
-                    stack.push(State.BOLD_END);
-                    return;
-                } else {
-                    handleBold(currentRoot);
-                }
+            if (currentToken.equals(Token.DoubleStar) && isBoldModusActive) {
+                break;
+            } else if (currentToken.equals(Token.DoubleStar)) {
+                handleBold(currentRoot);
+            } else if (currentToken.equals(Token.Star) && isItalicModusEnabled) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 130. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)